src/html_toast.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly/core';
+
+/**
+ * Configuration options for toasts.
+ */
+interface HtmlToastOptions extends Blockly.ToastOptions {
+  element?: HTMLElement;
+}
+
+/**
+ * Custom toast implementation that supports HTML elements in toast messages.
+ */
+class HtmlToast extends Blockly.Toast {
+  /**
+   * Creates the body of the toast for display.
+   *
+   * @param workspace The workspace the toast will be displayed on.
+   * @param options Configuration options for toast appearance/behavior.
+   * @returns The body for the toast.
+   */
+  protected static override createDom(
+    workspace: Blockly.WorkspaceSvg,
+    options: Blockly.ToastOptions,
+  ) {
+    const dom = super.createDom(workspace, options);
+    const contents = dom.querySelector('div');
+    if (
+      contents &&
+      'element' in options &&
+      options.element instanceof HTMLElement
+    ) {
+      contents.innerHTML = '';
+      contents.appendChild(options.element);
+    }
+    return dom;
+  }
+}
+
+/**
+ * Registers HtmlToast as the default toast implementation for Blockly. */
```

Comment:
Looks like there's a formatting issue here per the eslint error. Please fix.

Reply:
Done.